### PR TITLE
tests: add tests for nested multitype flag

### PR DESCRIPTION
**Pull request by anshumanv**
**What kind of change does this PR introduce?**

tests

**Did you add tests for your changes?**
yes

**If relevant, did you update the documentation?**
no

**Summary**
no

**Does this PR introduce a breaking change?**
no

**Other information**
no

### CODE REVIEW

File: packages/webpack-cli/__tests__/arg-parser.test.js [hunk 1]
```diff
@@ -448,4 +448,17 @@ describe('arg-parser', () => {
         });
         expect(warnMock.mock.calls.length).toEqual(0);
     });
+
+    it('parses nested multiType flag', () => {
+        const res = argParser(basicOptions, ['--env', 'a.b=d', '--env', 'a.b.c=d'], true);
```
Comment (snitin315) [inline packages/webpack-cli/__tests__/arg-parser.test.js:453]: Lets add one more case for 
```suggestion
        const res = argParser(basicOptions, ['--env', 'a.b=c', '--env', 'd.e.f=g'], true);
```

Comment (anshumanv) [reply packages/webpack-cli/__tests__/arg-parser.test.js:453]: yes

Comment (anshumanv) [reply packages/webpack-cli/__tests__/arg-parser.test.js:453]: we should also add multi compiler I think

Comment (anshumanv) [reply packages/webpack-cli/__tests__/arg-parser.test.js:453]: wip

Comment (anshumanv) [reply packages/webpack-cli/__tests__/arg-parser.test.js:453]: multicompiler not in scope, added more cases

**Review by alexander-akait — APPROVED** [review]
Looks good

**Review by snitin315 — APPROVED** [review]

**Review by rishabh3112 — APPROVED** [review]

**Review by jamesgeorge007 — APPROVED** [review]

**Pull request merged**